concrete/src/Cache/Adapter/ZendCacheDriver.php

```diff
@@ -27,27 +27,78 @@
 class ZendCacheDriver extends AbstractAdapter implements StorageInterface, FlushableInterface
 {
     /**
-     * @var string Name of the cache being used
+     * @var string Name of the cache being used when the localization is fully initialized
      */
-    private $cacheName;
+    private $finalCacheName;
+
+    /**
+     * @var \Concrete\Core\Application\Application
+     */
+    protected $app;
+
+    /**
+     * @var \Concrete\Core\Config\Repository\Repository
+     */
+    protected $config;
+
+    /**
+     * Is the localization system ready?
+     *
+     * @var bool
+     */
+    private $localizationReady;
 
     /**
      * @var int Number of seconds to consider the cache fresh before it expires
      */
     protected $cacheLifetime;
 
     /**
-     * @param string $cacheName Name of the cache being used. Defaults to cache.
-     * @param int $cacheLifetime Number of seconds to consider the cache fresh before it expires.
+     * @param string $finalCacheName Name of the cache being used when the localization is fully initialized. Defaults to cache
+     * @param int $cacheLifetime Number of seconds to consider the cache fresh before it expires
      */
-    public function __construct($cacheName = 'cache', $cacheLifetime = null)
+    public function __construct($finalCacheName = 'cache', $cacheLifetime = null)
     {
         parent::__construct();
-
-        $this->cacheName = $cacheName;
+        $this->setApplication(Application::getFacadeApplication());
+        $this->localizationReady = false;
+        $this->finalCacheName = $finalCacheName;
         $this->cacheLifetime = $cacheLifetime;
     }
 
+    public function setApplication(\Concrete\Core\Application\Application $app)
+    {
+        $this->app = $app;
+        $this->config = $app->make('config');
+    }
+
+    /**
+     * Is the localization system ready?
+     *
+     * @return bool
+     */
+    protected function isLocalizationReady()
+    {
+        if (!$this->localizationReady) {
+            if ($this->config->get('app.bootstrap.packages_loaded') === true) {
```

Comment:
I know why we do this, but just seeing `$this->config->get('app.bootstrap.packages_loaded')` here makes me sad. We should've made a `$status = $app->packageStatus();` or something.